beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -571,3 +577,62 @@ func (s *Server) RegisterValidator(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+// PrepareBeaconProposer endpoint saves the fee recipient given a validator index, this is used when proposing a block.
+func (s *Server) PrepareBeaconProposer(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.PrepareBeaconProposer")
+	defer span.End()
+
+	var jsonFeeRecipients []*shared.FeeRecipient
+	err := json.NewDecoder(r.Body).Decode(&jsonFeeRecipients)
+	switch {
+	case err == io.EOF:
+		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	case err != nil:
+		http2.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	var feeRecipients []common.Address
+	var validatorIndices []primitives.ValidatorIndex
+	// filter for found fee recipients
+	for i, r := range jsonFeeRecipients {
+		validatorIndex, valid := shared.ValidateUint(w, "Validator Index", r.ValidatorIndex)
+		if !valid {
+			return
+		}
+		feeRecipientBytes, err := hexutil.Decode(r.FeeRecipient)
```

Comment:
As discussed in publishBlockV3, we should probably return the hex from `shared.ValidateHex`